src/Common/Services/AppInstallationHandler.cs

```diff
@@ -1,25 +1,31 @@
-﻿using Cmf.CustomerPortal.BusinessObjects;
-using Cmf.CustomerPortal.Orchestration.CustomerEnvironmentManagement.InputObjects;
-using Cmf.MessageBus.Messages;
-using Newtonsoft.Json;
-using System;
+﻿using System;
 using System.IO;
+using System.Text.RegularExpressions;
 using System.Threading;
 using System.Threading.Tasks;
+using Cmf.CustomerPortal.BusinessObjects;
+using Cmf.CustomerPortal.Orchestration.CustomerEnvironmentManagement.InputObjects;
+using Cmf.MessageBus.Messages;
+using Newtonsoft.Json;
 
 namespace Cmf.CustomerPortal.Sdk.Common.Services
 {
     public class AppInstallationHandler : IAppInstallationHandler
     {
         private readonly ISession _session;
-
         private readonly ICustomerPortalClient _customerPortalClient;
-        
         private readonly IArtifactsDownloaderHandler _artifactsDownloaderHandler;
 
         private bool _isInstallationFinished = false;
-
         private bool _hasInstallationFailed = false;
+        private bool _hasInstallationStarted = false;
+        private readonly string[] loadingChars = { "|", "/", "-", "\\" };
+        private const string queuePositionMsg = "Queue Position:";
+        string pattern = @$"{queuePositionMsg} \d+\n";
+        private (int left, int top)? queuePositionCursorCoordinates = null;
+        private (int left, int top) queuePositionLoadingCursorCoordinates;
+        CancellationTokenSource cancellationTokenDeploymentQueued;
+        static SemaphoreSlim semaphore = new(1);
```

Comment:
Does this need to be static?

Reply:
i think yes, otherwise will not be the same instance of Semaphore shared for each thread